Sage/carbon
# fix(menu-fullscreen): add listener to trigger focus trap on transitionend event FE-4252

fix #4287

### Proposed behaviour


Adds `eventListener` to trigger `FocusTrap` on `transitionend` event, focus is set on the content wrapper rather than a `MenuItem`/ `IconButton`. Adds handle for escape key press closing menu as well

### Current behaviour


Focus is not trapped on open on `MenuFullscreen`

### Checklist




#### QA


### Additional context

![menu-fullscreen-focus-trap](https://user-images.githubusercontent.com/44157880/136542921-cf14e8f4-d9bb-4d14-9365-4d55cdb7ee9d.gif)

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
Behaviour should be testable in `menu--fullscreen-view` story
